docker/py/Dockerfile

```diff
@@ -1,5 +1,37 @@
-ARG BASE_IMAGE=jupyter/base-notebook:lab-3.4.0
-FROM $BASE_IMAGE as base
+ARG BASE_IMAGE=jupyter/base-notebook:python-3.10
+
+FROM ${BASE_IMAGE} as builder
+
+USER root
+RUN apt-get update && apt-get install -yq --no-install-recommends \
+    build-essential && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# switch to the notebook user
+USER $NB_USER
+# install jupyterlab, papermill, git extension and renku-jupyterlab-ts
+COPY requirements.txt /tmp/requirements.txt
+RUN python3 -m pip install --no-cache-dir -U pip && \
+    python3 -m pip install --no-cache-dir -r /tmp/requirements.txt && \
+    # jupyter lab build && \
+    # jupyter labextension list && \
+    # npm cache clean --force && \
+    rm -rf "/home/${NB_USER}/.cache"
+
+# jupyter sets channel priority to strict which often causes very long error messages
+RUN mamba config --system --set channel_priority flexible && \
+    mamba clean --all -f -y
+
+COPY renku-requirements/requirements.txt /tmp/renku-requirements.txt
+
+RUN mkdir -p "$HOME/.renku/bin" && \
```

Comment:
@rokroskar why dont we move the linux package installation in the base image here? I know we discussed this briefly but how did you decide what goes in the first stage?

Reply:
Nvm I remembered our discussion. You just want to skip the python env setup with the build stage. But all other things are installed in the final stage.

Reply:
Only the python install is done here - the apt is unfortunately needed to install the compiler (there is no arm wheel for psutil afaict)

Reply:
basically we can copy the entire content of what gets installed for the python dependencies because it all goes into the conda directory. The rest is not easy to transfer between stages. I thought about streamlining the big apt install (we probably don't _need_ all of that) but I left it for another day.